### PR TITLE
Update README.md and obs script

### PR DESCRIPTION
**Pull request by aaronspring**
- [ ] more explicit describing, see bjorns comment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,8 +93,10 @@ The forecast start dates in this dataset are from 2020/01/02 to 2020/12/31.
 The `hindcast-like-observations` (`training-output-reference`) dataset.
 The `forecast-like-observations` (`test-output-reference`) dataset.
 
-The observations are the ground truth to compare with the ML model output and evaluate them. It consists in observation from instruments of temperature and accumulated total precipitation. (TODO add more descriptions) (point to the scripts to create them ? TODO).
-Generally speaking, only past data can be used by the ML models to perform their forecast :
```
Comment (aaronspring) [inline README.md:95]: they can also use the forecasts issued on the same date. see my suggestion below

File: README.md [hunk 2]
```diff
@@ -36,63 +37,66 @@ Fostering discussions about how to prevent overfitting may be an outcome of the
...
+The observations are the ground truth to compare with the ML model output and evaluate them. It consists in observation from instruments of [temperature](http://iridl.ldeo.columbia.edu/SOURCES/.NOAA/.NCEP/.CPC/.temperature/.daily/) and accumulated total [precipitation](http://iridl.ldeo.columbia.edu/SOURCES/.NOAA/.NCEP/.CPC/.UNIFIED_PRCP/.GAUGE_BASED/.GLOBAL/.v1p0/.extREALTIME/.rain/). The [NOAA CPC](https://www.cpc.ncep.noaa.gov/) datasets were downloaded from [IRIDL](iridl.ldeo.columbia.edu/). We provide observations in the same dimensions as the forecasts/hindcasts to have an easy match of forecasts/hindcast and ground truth. [See the script for technical details](https://github.com/ecmwf-lab/climetlab-s2s-ai-challenge/tree/main/tools/observations).
+
+
+Generally speaking, only data available when the forecast is issued can be used by the ML models to perform their forecast:
 
 __Rule 1 : Observed data beyond the forecast date should not be used for prediction, for instance a forecast starting on 2020/07/01 should not use observed data beyond 2020/07/01).__
```
Comment (aaronspring) [inline README.md:101]: maybe we could link here to the rules: https://s2s-ai-challenge.github.io/#rules

File: README.md [hunk 3]
```diff
@@ -132,20 +137,22 @@ The URLs to download the data are constructed according to the following pattern
...
-- {origin} : "ecmwf" or "eccc" or "ncep".
-- {fctype} : "hindcast" ("forecast" for test dataset).
-- {weeks} is "34" or "56".
-- YYYYMMDD is the date of main forecast time in the file.
+
+- {datasetname} : In the URLs the dataset name must follow the ML naming (`training-input` or `training-output-reference` or `training-output-benchmark`).
+- {format} is `netcdf`. Training output is also available as GRIB file,  using `format='grib'` and replacing `".nc"` by `".grib"`
+- {parameter} is `t2m` for surface temperature at 2m, "tp" for total precipitation using CF convention.
+- {origin} : `ecmwf` or `eccc` or `ncep` <!-- we should have a clean table for this once -->
+- {fctype} : "hindcast" ("forecast" for test dataset). <!-- is this needed in the API? IMO not -->
```
Comment (aaronspring) [inline README.md:145]: can we delete `fctype`? this is not needed anymore, at least in the external API? @floriankrb

File: README.md [hunk 4]
```diff
@@ -132,20 +137,22 @@ The URLs to download the data are constructed according to the following pattern
...
-- {format} is "netcdf". Training output is also available as GRIB file,  using format=grib and replacing ".nc" by ".grib"
-- {parameter} is "t2m" for surface temperature at 2m, "tp" for total precipitation using CF convention.
-- {origin} : "ecmwf" or "eccc" or "ncep".
-- {fctype} : "hindcast" ("forecast" for test dataset).
-- {weeks} is "34" or "56".
-- YYYYMMDD is the date of main forecast time in the file.
+
+- {datasetname} : In the URLs the dataset name must follow the ML naming (`training-input` or `training-output-reference` or `training-output-benchmark`).
+- {format} is `netcdf`. Training output is also available as GRIB file,  using `format='grib'` and replacing `".nc"` by `".grib"`
+- {parameter} is `t2m` for surface temperature at 2m, "tp" for total precipitation using CF convention.
```
Comment (aaronspring) [inline README.md:143]:
```suggestion
- {parameter} is `t2m` for [surface temperature at 2m](https://confluence.ecmwf.int/display/S2S/S2S+Surface+Air+Temperature), `tp` for [total precipitation](https://confluence.ecmwf.int/display/S2S/S2S+Total+Precipitation)
```

File: README.md [hunk 5]
```diff
@@ -132,20 +137,22 @@ The URLs to download the data are constructed according to the following pattern
...
-- {origin} : "ecmwf" or "eccc" or "ncep".
-- {fctype} : "hindcast" ("forecast" for test dataset).
-- {weeks} is "34" or "56".
-- YYYYMMDD is the date of main forecast time in the file.
+
+- {datasetname} : In the URLs the dataset name must follow the ML naming (`training-input` or `training-output-reference` or `training-output-benchmark`).
+- {format} is `netcdf`. Training output is also available as GRIB file,  using `format='grib'` and replacing `".nc"` by `".grib"`
+- {parameter} is `t2m` for [surface temperature at 2m](https://confluence.ecmwf.int/display/S2S/S2S+Surface+Air+Temperature), `tp` for [total precipitation](https://confluence.ecmwf.int/display/S2S/S2S+Total+Precipitation)
+- {origin} : `ecmwf` or `eccc` or `ncep` <!-- we should have a clean table for this once -->
+- {fctype} : "hindcast" ("forecast" for test dataset). <!-- is this needed in the API? IMO not -->
```
Comment (aaronspring) [inline README.md:145]:
```suggestion
```

File: README.md (hunk 2 above)
Comment (aaronspring) [inline README.md]: this sounds like there are also other rules here? I think you had here more in the past, but I see this now as the only rule.

Comment (floriankrb) [reply README.md]: ok,. README updated.

File: README.md [hunk 6]
```diff
@@ -7,23 +7,24 @@
...
+There are four datasets provided for this challenge. As we are aiming at bringing together the two communities of Machine Learning and Weather Prediction, they have been aliases to use both two points of views:
 
 | ML                          | NWP                          |                                                        |
 | --------------------------- | ---------------------------- | ------------------------------------------------------ |
 | `training-input`            | `hindcast-input`             | Training  dataset (input for training the ML models)   |
 | `training-output-reference` | `hindcast-like-observations` | Training dataset (output for training the ML models)   |
 | `training-output-benchmark` | `hindcast-benchmark`         | Benchmark output (on the training dataset) (NOT YET AVAILABLE)     |
-| `test-input`                | `forecast-input`             | Test dataset (DO NOT use)                              |
+| `test-input`                | `forecast-input`             | Test dataset (DO NOT use for training)                              |
```
Comment (floriankrb) [inline README.md:23]: No. Do not use for training the ML model, but do not use to design them. Do no use to talk about performance of ML model. Do not use to have ideas about what to do. Do not use it during the challenge.

**Review by floriankrb — APPROVED** [review]
I commented a few points, all the other changes are good for me.

File: README.md [hunk 7]
```diff
@@ -36,65 +37,70 @@ Fostering discussions about how to prevent overfitting may be an outcome of the
...
 These data are forecast data.
 This could be used the input for applying the ML models in order to generate the output which is submitted for the challenge.
 Using data from earlier date that 2020/01/01 is also allowed during the prediction phase.
 The forecast start dates in this dataset are from 2020/01/02 to 2020/12/31.
-  - For all 3 models : 
-    - forecast_time : from 2020/01/02 to 2020/12/31, weekly every 7 days (every Thurday).
-    - valid_time (forecast_time + lead_time): from 2020/01/02 to 2020/12/31
+  - For all 3 models: 
+    - `forecast_time`: from 2020/01/02 to 2020/12/31, weekly every 7 days (every Thurday).
+    - `valid_time` (`forecast_time` + `lead_time`): from 2020/01/02 to 2021/02/xx <!-- valid_time extends into 2021 -->
```
[the remark this answers is not in the document]
Comment (aaronspring) [reply README.md:76]: doesnt it depend on the model as they have different leads? therefore I wrote xx as a placeholder

File: README.md (hunk 6 above)
Comment (aaronspring) [reply README.md:23]: Dont they need the test-input to create their ML-based predictions? I think they need and thats why it is called forecast-input.

File: README.md [hunk 8]
```diff
@@ -36,65 +37,70 @@ Fostering discussions about how to prevent overfitting may be an outcome of the
...
 ### Hindcast input (Training input)
 
-The `hindcast-input`(`training-input`) dataset consists in data from three different models : ECMWF (ecmf), ECCC (cwao), NCEP (kwbc).
+The `hindcast-input`(`training-input`) dataset consists of data from three different models: ECMWF (ecmf), ECCC (cwao), NCEP (kwbc).
+Use `origin="ecmwf"` (model name) or `origin="ecmwf"` (center name).
 These data are hindcast data. This is used as the input for training the ML models.
-This dataset is available as grib, netcdf or zarr.
-In this dataset, the data is available from 1998 for the oldest, to 2019/12/31 for the most recent. 
+This dataset is available as `grib`, `netcdf` or `zarr`.
+In this dataset, the data is available from 1999 for the oldest, to 2019/12/31 for the most recent. <!-- which model is available from 1998? -->
```
[the remark this answers is not in the document]
Comment (aaronspring) [reply README.md:44]: so 1999 is fine.

File: README.md [hunk 9]
```diff
@@ -36,65 +37,70 @@ Fostering discussions about how to prevent overfitting may be an outcome of the
...
-    - forecast_time : from , weekly every 7 days (every Thurday).
-    - lead_time : 1 to 32 days
-    - valid_time (forecast_time + lead_time): from 
-    - variables sm20, sm100, st20, st100 not available
-    - availables parameters : 2t/ci/gh/lsm/msl/q/rsn/sp/sst/t/tcc/tcw/tp/ttr/u/v
+    - `forecast_time`: from to , weekly every 7 days (every Thurday).
+    - `lead_time`: 1 to 32 days
+    - `valid_time` (forecast_time + lead_time): from 
+    - availables parameters: `2t/ci/gh/lsm/msl/q/rsn/sp/sst/t/tcc/tcw/tp/ttr/u/v`
+    - parameter not available: sm20, sm100, st20, st100
```
Comment (aaronspring) [inline README.md:55]:
```suggestion
    - parameters not available: sm20, sm100, st20, st100
```

File: README.md [hunk 10]
```diff
@@ -111,16 +117,17 @@ The observations dataset have been build from real instrument observations.
...
 
 The benchmark data is available as follows :
-  - forecast_time : from 2020/01/01 to 2020/12/31, weekly every 7 days (every Thurday).
-  - lead_time : two values : 28 days and 35 days (To be discussed)
-  - valid_time (forecast_time + lead_time): from 2020/01/01 to 2020/12/31
-  - category : tercile (TODO document this better)
+  - `forecast_time`: from 2020/01/02 to 2020/12/31, weekly every 7 days (every Thurday).
+  - `lead_time`: 14 days and 28 days, where this day represents the first day of the biweekly aggregate
+  - `valid_time` (`forecast_time` + `lead_time`): from 2020/01/01 to 2021/01/29
+  - `category`: `'below normal'`, `'near normal'`, `'above normal'` <!-- todo: implement in stoarge and/or climetlab @florian -->
```
[the remark this answers is not in the document]
Comment (aaronspring) [reply README.md:130]: for now I hard code this in my example

[the remark this answers is not in the document]
Comment (aaronspring) [reply README.md:130]: this todo is not visible. its a comment

**Comment by aaronspring** [comment]
can you resolve the conflicts @floriankrb or shall I?

**Pull request merged**